src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -869,7 +869,7 @@ public void Close()
 
                     if (_standardError != null && (_errorStreamReadMode == StreamReadMode.AsyncMode || _errorStreamReadMode == StreamReadMode.Undefined))
                     {
-                        if (_errorStreamReadMode == StreamReadMode.AsyncMode)
+                        if (_error != null && _errorStreamReadMode == StreamReadMode.AsyncMode)
```

Comment:
These could be done instead just by changing the . to ?. on the invocation below.